raster/src/main/scala/geotrellis/raster/summary/polygonal/PolygonalSummaryResult.scala

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2019 Azavea
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package geotrellis.raster.summary.polygonal
+
+import cats.Monad
+
+import scala.annotation.tailrec
+
+/**
+  * A Result ADT returned by [[PolygonalSummary]] operations
+  *
+  * There are two result types, [[NoIntersection]] and [[Summary]].
+  * This ADT will always return Summary if the Raster and Polygon intersect.
+  * Otherwise, NoIntersection will be returned.
+  *
+  * Provides helpers to convert to Option and Either if you don't need to match
+  * on result type.
+  */
+sealed trait PolygonalSummaryResult[+A] {
+  def toOption: Option[A]
+
+  def toEither: Either[Any, A]
+
+  implicit val noIntersectionMonad: Monad[PolygonalSummaryResult] = new Monad[PolygonalSummaryResult] {
```

Comment:
A better place for it in `PolygonalSummaryResult` companion object.

Reply:
Ha yeah I realized this later, this change is in the Spark API companion PR over at cloudniner/geotrellis#2.